theseus_gui/src-tauri/src/api/auth.rs

```diff
@@ -1,56 +1,73 @@
 use crate::api::Result;
-use theseus::prelude::*;
+use theseus::{prelude::*, window_scoped};
 
 /// Authenticate a user with Hydra - part 1
 /// This begins the authentication flow quasi-synchronously, returning a URL to visit (that the user will sign in at)
 #[tauri::command]
-pub async fn auth_authenticate_begin_flow() -> Result<url::Url> {
-    Ok(auth::authenticate_begin_flow().await?)
+pub async fn auth_authenticate_begin_flow(
+    window: tauri::Window,
+) -> Result<url::Url> {
+    Ok(window_scoped!(window, auth::authenticate_begin_flow()).await?)
```

Comment:
what is window scoped used for? does it need to be used for every single command?